GitCommands/Git/GitModule.cs

```diff
@@ -835,6 +835,25 @@ private IGitItem GetSubmoduleCommitHash(string filename, string refName)
             return null;
         }
 
+        public (int? first, int? second) GetCommitRangeDiffCount(ObjectId firstId, ObjectId secondId)
+        {
+            var args = new GitArgumentBuilder("rev-list")
+            {
+                $"{firstId}...{secondId}",
+                "--count",
+                "--left-right"
+            };
```

Comment:
Could we use constants for these commands?

```
public static class GitCommands
{
    public static class RevList
    {
        public const string Command = "rev-list";
        public const string CountParameter = "--count";
        public const string LeftRightParameter = "--left-right";
    }
}
```

or

```
public static class GitCommands
{
    public const string RevList = "rev-list";

    public static class RevListParameters
    {
        public const string Count = "--count";
        public const string LeftRight = "--left-right";
    }
}
```

Most likely there will be many common parameters.

Reply:
If this is changed, it should be a generic change for all commands, not in this PR.

Personally I see just one reason to make constants for commands: It is possible to search where they are used.
"rev-list" is unique though, but "log" is not.

Other than that, for Git commands you get down to a text string anyway. Using constants will just add another layer when viewing the code.

Other than that, I would like to merge this.
Changes to IGitModule will affect more commands than GetCommitRangeDiffCount, like GetDiffFilesWithSubmodulesStatus.

Reply:
> Personally I see just one reason to make constants for commands: It is possible to search where they are used.
"rev-list" is unique though, but "log" is not.

The search itself is important.

I use the rule: "Everything that can be calculated at the stage of creating a program must be calculated."
Example:

![image](https://user-images.githubusercontent.com/3169265/100396921-e0fa9400-304f-11eb-914d-98f4debd3660.png)

![image](https://user-images.githubusercontent.com/3169265/100396949-fff92600-304f-11eb-909c-aa83452f1f9a.png)

> Other than that, for Git commands you get down to a text string anyway. Using constants will just add another layer when viewing the code.

And that's the key.
Why not use it like that?

`var args = new GitArgumentBuilder($"rev-list {firstId}...{secondId} --count --left-right");`

Or so?

`var args = $"rev-list {firstId}...{secondId} --count --left-right";`

I think we could find this door if we looked, the key is already in our hands. 😉

> Other than that, I would like to merge this.

Well, Ok.

Reply:
> Why not use it like that?

The Git command (like "rev-list") is a hook for options etc so that must be separate.
The (generic) argument builder handles some types native and simplifies use of booleans etc.

Another layer will require me too look up the constant when browsing the code which is not an advantage to me.

Reply:
```
var args = new GitArgumentBuilder("rev-list");
var args = new GitArgumentBuilder(GitCommands.RevList);
```

I understand correctly, do you think that the second method may require additional actions?

Reply:
> I understand correctly, do you think that the second method may require additional actions?

No additional actions, but I do not see the actual commands at a glance.